sdk/identity/azure-identity/samples/client_certificate_credential.md

```diff
@@ -0,0 +1,70 @@
+# Using the CertificateCredential
+
+Applications which execute in a protected environment can authenticate using a client assertion signed by a private key whose public key or root certificate is registered with AAD. The Azure.Identity library provides the `CertificateCredential` for applications choosing to authenticate this way. Below are some examples of how applications can utilize the `CertificateCredential` to authenticate clients.
+
+
+## Loading certificates from disk
+
+Applications commonly need to load a client certificate from disk. One approach is for the application to construct the `CertificateCredential` by specifying the applications tenant id, client id, and the path to the certificate.
+
+```py
+credential = CertificateCredential(tenant_id, client_id, "./certs/cert.pfx")
+```
+Alternatively, the application can load the certificate themselves, such as in the following example.
+
+```C# Snippet:Identity_CertificateCredenetial_CreateWithX509Cert
+certificate_data = open(CERT_PATH, "rb").read()
+
+credential = new CertificateCredential(tenant_id, client_id, certificate_data=certificate_data);
+```
+
+## Rolling Certificates
+
+Long running applications may have the need to roll certificates during process execution. Certificate rotation is not currently supported by the `CertificateCredential` which treats the certificate used to construct the credential as immutable. This means that any clients constructed with a `CertificateCredential` using a particular cert would fail to authenticate requests after that cert has been rolled and the original is no longer valid. 
+
+However, if an application wants to roll this certificate without creating new service clients, it can accomplish this by creating its own `TokenCredential` implementation which wraps the `CertificateCredential`. The implementation of this custom credential `TokenCredential` would somewhat depend on how the application handles certificate rotation.
+
+### Explicit rotation
+
+If the application get's notified of certificate rotations and it can directly respond, it might choose to wrap the `CertificateCredential` in a custom credential which provides a means for rotating the certificate. 
```

Comment:
```suggestion
If the application gets notified of certificate rotations and it can directly respond, it might choose to wrap the `CertificateCredential` in a custom credential which provides a means for rotating the certificate. 
```